app/controllers/tasks_controller.rb

```diff
@@ -13,35 +13,110 @@ class TasksController < ApplicationController
   end
 
   post '/tasks' do
-    task = Task.new(description: params[:description])
-
-    if task.save
-      redirect "/"
+    if is_json_request?
+	request_body = request.body.read
```

Comment:
This functionality is already baked in, so you don't need to be parsing the request body yourself.